bridge-token-factory/src/lib.rs

```diff
@@ -413,7 +753,62 @@ impl BridgeTokenFactory {
         );
         let token_address = validate_eth_address(parts[0].to_string());
         let recipient_address = validate_eth_address(recipient);
-        result_types::Withdraw::new(amount, token_address, recipient_address)
+
+        let withdraw_fee_bound = self.get_withdraw_token_fee_bound(&parts[0].to_string());
+        let withdraw_fee_percentage = self
+            .get_withdraw_token_fee_percentage(&parts[0].to_string())
+            .unwrap_or(WithdrawFeePercentage {
+                near_to_eth: 0,
+                aurora_to_eth: 0,
+            });
+
+        let amount_to_transfer: u128;
+        let mut fee_amount: u128;
+        // check whether the withdrawer is aurora native account or it's silos
+        if is_aurora_engine_account(withdrawer.clone()) {
+            let silo_fee = self
+                .get_withdraw_fee_percentage_per_silo_per_token(&withdrawer, &parts[0].to_string());
+            if silo_fee != 0 {
+                fee_amount = (amount * silo_fee) / FEE_DECIMAL_PRECISION;
+            } else {
+                fee_amount =
+                    (amount * withdraw_fee_percentage.aurora_to_eth) / FEE_DECIMAL_PRECISION;
+            }
+        } else {
+            fee_amount = (amount * withdraw_fee_percentage.near_to_eth) / FEE_DECIMAL_PRECISION;
+        }
+        // If some token bounds is set for withdraw than update the fee amount according to bounds
+        match withdraw_fee_bound {
+            Some(token_bounds) => {
+                if fee_amount < token_bounds.lower_bound {
+                    //bound checks
+                    fee_amount = token_bounds.lower_bound;
+                } else if fee_amount > token_bounds.upper_bound {
+                    fee_amount = token_bounds.upper_bound;
+                }
+            }
+            _ => {}
+        }
```

Comment:
This code could be moved to the separate function `calculate_withdraw_fee`

Reply:
And the same for the deposit

Reply:
made a generic function which uses enum. e53607f60125d25cb40abd0482f1120da1552d62

Reply:
It is not enough, all the fee calculations should be moved to separate methods, so the top-level function should contain just one line for the fee `let fee_amount = get_withdraw_fee(...)`

Reply:
382d2c2e632f5db6943c4753a9c7c8f397f411cd